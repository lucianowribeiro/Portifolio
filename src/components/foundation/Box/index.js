import styled from 'styled-components';
import propToStyle from '../../../theme/utils/propToStyle';

const Box = styled.div`
  ${propToStyle('display')}
  ${propToStyle('flex')}
  ${propToStyle('flexDirection')}
  ${propToStyle('justifyContent')}
  ${propToStyle('marginBottom')}
  ${propToStyle('alignItems')}
  ${propToStyle('backgroundColor')}
  ${propToStyle('boxShadow')};
  ${propToStyle('width')}
  ${propToStyle('height')}
  ${propToStyle('borderRadius')}
  ${propToStyle('listStyle')}
  ${propToStyle('marginBottom')}
  ${propToStyle('paddingLeft')}
  ${propToStyle('textAlign')}
`;
export default Box;
