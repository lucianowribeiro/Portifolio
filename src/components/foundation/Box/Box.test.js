import React from 'react';
import { create } from 'react-test-renderer';
import Box from './index';

describe('<Box />', () => {
  describe('it renders', () => {
    test('without props and children', () => {
      expect(create(<Box />).toJSON()).toMatchSnapshot();
    });
    test('with element chlidren centered in the middle of the screen', () => {
      const childrenMock = 'Teste';
      const component = create(
        <Box
          display="flex"
          justifyContent="center"
          alignItems="center"
          width="30%"
          height="30%"
        >
          {childrenMock}
        </Box>,
      ).toJSON();
      // Assertations
      expect(...component.children).toBe(childrenMock);
      expect(component).toHaveStyleRule('width', '30%');
      expect(component).toHaveStyleRule('height', '30%');
      expect(component).toHaveStyleRule('display', 'flex');
      expect(component).toHaveStyleRule('justify-content', 'center');
      expect(component).toHaveStyleRule('align-items', 'center');
      // Snapshot
      expect(component).toMatchSnapshot();
    });
  });
});
