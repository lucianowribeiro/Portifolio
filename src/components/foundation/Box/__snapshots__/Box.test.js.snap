// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> it renders with element chlidren centered in the middle of the screen 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 30%;
  height: 30%;
}

<div
  className="c0"
  display="flex"
  height="30%"
  width="30%"
>
  Teste
</div>
`;

exports[`<Box /> it renders without props and children 1`] = `
<div
  className=""
/>
`;
